Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+/**
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include "SwmmInterface.h"
+
+#include <utility>
+
+#include "BaseLib/FileTools.h"
+#include "BaseLib/StringTools.h"
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/Point.h"
+#include "GeoLib/Polyline.h"
+#include "GeoLib/Polygon.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Line.h"
+#include "MeshLib/Properties.h"
+
+#include "Applications/FileIO/CsvInterface.h"
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+const std::array<std::string,9> subcatchment_vars =
+{
+    "rainfall",
+    "snow depth",
+    "evaporation loss",
+    "infiltration losses",
+    "runoff rate",
+    "groundwater outflow",
+    "groundwater head",
+    "moisture content",
+    "concentration of pollutant"
+};
+
+const std::array<std::string,7> node_vars =
+{
+    "water depth",
+    "hydraulic head",
+    "volume of stored water",
+    "lateral inflow",
+    "total inflow",
+    "flow lost to flooding",
+    "concentration of pollutant"
+};
+
+const std::array<std::string,6> link_vars =
+{
+    "flow rate",
+    "flow depth",
+    "flow velocity",
+    "flow volume",
+    "fraction conduit/non-conduit",
+    "concentration of pollutant"
+};
+
+const std::array<std::string,15> system_vars =
+{
+    "air temperature",
+    "rainfall",
+    "snow depth",
+    "evaporation + infiltration loss",
+    "runoff flow",
+    "dry weather inflow",
+    "groundwater inflow",
+    "RDII inflow",
+    "direct inflow",
+    "total lateral inflow",
+    "flow lost to flooding",
+    "flow leaving through outfalls",
+    "volume of stored water",
+    "actual evaporation rate",
+    "potential evaporation rate"
+};
+
+std::array<std::size_t,4> const n_obj_params = { 8, 6, 5, 15 };
+
+SwmmInterface* SwmmInterface::create(std::string const& file_name)
+{
+    if (file_name.length() < 5)
```

Comment:
The input/output methods take a base name and check if the corresponding i/o file for that base name exists. This check takes any swmm project file, i.e. [base name].[extension] which needs to be at least 5 chars because the extension is always 3 chars.